shell/browser/net/electron_url_loader_factory.cc

```diff
@@ -312,17 +325,20 @@ void ElectronURLLoaderFactory::StartLoading(
     case ProtocolType::kFree:
       ProtocolType type;
       if (!gin::ConvertFromV8(args->isolate(), response, &type)) {
-        mojo::Remote<network::mojom::URLLoaderClient> client_remote(
-            std::move(client));
-        client_remote->OnComplete(
-            network::URLLoaderCompletionStatus(net::ERR_FAILED));
+        OnComplete(std::move(client), request_id,
+                   network::URLLoaderCompletionStatus(net::ERR_FAILED),
+                   proxy_factory);
         return;
       }
       StartLoading(std::move(loader), routing_id, request_id, options, request,
                    std::move(client), traffic_annotation, proxy_factory, type,
                    args);
       break;
   }
+
+  if (type != ProtocolType::kFree && proxy_factory) {
```

Comment:
It is not very straightforward why the `type != ProtocolType::kFree` check is needed, how about doing `return` instead of `break` in `case ProtocolType::kFree`? It should make the code more easy to understand.